File: Algorithm/leetcode_690.go
/**
 * File              : leetcode_690.go
 * Author            : Andy
 * Date              : 2022.10.11
 * Last Modified Date: 2022.10.11
 * Last Modified By  : Andy
 */

package getImportance

func getImportance(employees []*Employee, id int) int {
	var hashmap map[int]*Employee
	total := 0
	stack := []int{id}
	for i := 0; i < len(employees); i++ {
		hashmap[employees[i].Id] = employees[i]
	}

	for len(stack) != 0 {
		cur := stack[len(stack)-1]
		stack = stack[:len(stack)-1]

		total += hashmap[cur].Importance
		for _, sub := range hashmap[cur].Subordinates {
			stack = append(stack, sub)
		}
	}
	return total
}

File: Algorithm/leetcode_690.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# File              : leetcode_690.py
# Author            : Andy
# Date              : 2022.10.11
# Last Modified Date: 2022.10.11
# Last Modified By  : Andy

# 子元素的id可以通过subordinates获取到，直接遍历就行，
# 前提通过id获取到employee本身


class Solution:

    def getImportance(self, employees, id):
        total = 0
        stack = [id]
        hashmap = {}
        for e in employees:
            hashmap[e.id] = e

        while stack:
            cur = stack.pop()
            total += cur.importance

            for sub in subordinates:
                stack.append(sub)

        return total
